listenbrainz/webserver/views/playlist.py

```diff
@@ -0,0 +1,49 @@
+import ujson
+from werkzeug.exceptions import NotFound, BadRequest
+
+from flask import Blueprint, render_template, current_app
+from flask_login import current_user
+from listenbrainz.domain import spotify
+from listenbrainz.webserver.views.api_tools import is_valid_uuid
+from listenbrainz.webserver.views.playlist_api import serialize_jspf, fetch_playlist_recording_metadata
+import listenbrainz.db.playlist as db_playlist
+
+playlist_bp = Blueprint("playlist", __name__)
+
+
+@playlist_bp.route("/<playlist_mbid>", methods=["GET"])
+
+def load_playlist(playlist_mbid: str):
+    """Load a single playlist by id
+    """
+    if not is_valid_uuid(playlist_mbid):
+        raise BadRequest("Provided playlist ID is invalid: %s" % playlist_mbid)
+
+    playlist = db_playlist.get_by_mbid(playlist_mbid, True)
+    if playlist is None or \
+        (playlist.creator_id != current_user.get_id() and not playlist.public):
+        raise NotFound("Cannot find playlist: %s" % playlist_mbid)
+
+    fetch_playlist_recording_metadata(playlist)
+
+    spotify_data = {}
+    current_user_data = {}
+    if current_user.is_authenticated:
```

Comment:
Does this still error if not logged in?

Reply:
I seem to be able to access a public playlist in an incognito browser window (consequently not logged in): https://test.listenbrainz.org/playlist/10d6423e-8ff7-40f6-813d-5552753715d7

Reply:
It says `Public playlist by mr_monkey`

Reply:
I am unable to create a private playlist. It says 
![image](https://user-images.githubusercontent.com/33172244/101237218-fb420b00-36fc-11eb-854e-2e709c8ee95e.png)

Reply:
confirmed